005_team_3_agent_design/005_team_3_agent_design.tex

```diff
@@ -1 +1,129 @@
+    \item \texttt{morality int}: defines the willingness of the agent to help others, \cite{waller_2015}, in other words, how much you care about other agents in the tower. 
+    \item \texttt{mood int}: defines how likely the agent is to do things, it will affect its decision-making process.
+\end{enumerate}
+For the purpose of this agent, we will define personality as the individual differences in characteristic patterns of thinking, feeling, and behaving, as defined by the American Psychology Association. The traits that define an agent's personality are mutable, making our agent adapt its personality depending on the circumstances. 
+
+\subsection{Agent generation}
+%%TODO: have to add the code snippet
+The function generates a new agent by initializing its variables, knowledge and decisions. All knowledge and decisions are defined initially as empty except for our last HP, which is initialized as 100 (as all agents are). \par
+The variables of our agent, which define the personality it has, are randomly allocated to generate different agents. All three variables are defined to be in the range 0 to 100. Therefore morality and mood are randomly allocated to a value between 0 and 100. To ensure our agent is not completely deaf from the start we limit the initial stubbornness to 75. This does not limit the agent once it starts interacting and taking decisions. Which means it could possibly end up with a stubbornness value of 100 if the situations it goes through are auspicious for it.
+
+\subsection{Agent Strategy}
+Our agent follows the following strategy depending on the three agent variables:
+\begin{enumerate}
+    \item When morality is high, agents are more willing to listen and obey other agents.
+    \item When morality is low, agents are less willing to listen and choose to eat more than they would when they have high morality.
+    \item When stubbornness is high, agents will be “deaf” to most messages and not react to them.
+    \item When stubbornness is high, agents are more willing to listen to messages.
+    \item When mood is high, the actions taken are beneficial for more people.
+    \item When mood is low, the actions taken are more beneficial for ourselves.
+\end{enumerate}
+
+\section{Agents emotions}\label{agents_emotions}
+The following functions act as helpers for the rest of the agent code, enabling changes in the agents variables and reactions due to the value of these variables, as well as reactions to friendships. 
+
+\subsection{Read function}
+\texttt{func (a *CustomAgent3) read() bool}: our agent's stubbornness will limit the amount of messages it might read and react to. Each time we receive a message we call the \texttt{read()} function. This function will take into account the current stubbornness of the agent and return if the message must be read, and reacted to, or just discarded. 
```

Comment:
```suggestion
\texttt{func (a *CustomAgent3) read() bool}: our agent's stubbornness will limit the amount of messages it might read and react to. Each time an agent receives a message we call the \texttt{read()} function. This function will take into account the current stubbornness of the agent and return if the message must be read, and reacted to, or just discarded. 
```

"We" should refer to the programmer (you guys!), not the agent :)